pkg/collect/cluster_resources.go

```diff
@@ -613,29 +667,76 @@ func jobs(ctx context.Context, client *kubernetes.Clientset, namespaces []string
 }
 
 func cronJobs(ctx context.Context, client *kubernetes.Clientset, namespaces []string) (map[string][]byte, map[string]string) {
+	ok, err := discovery.HasResource(client, "batch.k8s.io/v1", "CronJobs")
+	if err != nil {
+		return nil, map[string]string{"": err.Error()}
+	}
+	if ok {
+		return cronJobsV1(ctx, client, namespaces)
+	}
+
+	return cronJobsV1beta(ctx, client, namespaces)
+}
+
+func cronJobsV1(ctx context.Context, client *kubernetes.Clientset, namespaces []string) (map[string][]byte, map[string]string) {
+	cronJobsByNamespace := make(map[string][]byte)
+	errorsByNamespace := make(map[string]string)
+
+	for _, namespace := range namespaces {
```

Comment:
I wonder if we can DRY this code a bit. If its non-trivial then just add a TODO for to have it looked at as tech-debt

Reply:
I've added a TODO comment for function `pdbV1`.